src/main/kotlin/game/racingcar/RacingCarGame.kt

```diff
@@ -1,35 +1,33 @@
 package game.racingcar
 
 import game.racingcar.dto.LocationSnapshot
+import game.racingcar.dto.RacingCarPlayResult
 import game.racingcar.move.MoveStrategy
 import game.racingcar.move.RandomMoveStrategy
 import game.racingcar.view.InputView
 import game.racingcar.view.OutputView
 
 class RacingCarGame(
-    private val numberOfCars: Int,
+    private val carNames: List<String>,
     private val numberOfTrials: Int,
     private val moveStrategy: MoveStrategy = RandomMoveStrategy()
 ) {
-    fun play(): List<LocationSnapshot> {
+    fun play(): RacingCarPlayResult {
         val locationSnapshots = mutableListOf<LocationSnapshot>()
 
-        val racingCars = RacingCars(numberOfCars, moveStrategy)
-        repeat(numberOfTrials) {
-            with(racingCars) {
-                moveAll()
-                locationSnapshots.add(LocationSnapshot(locations()))
-            }
+        val racingCars = RacingCars(carNames, moveStrategy)
+        (1..numberOfTrials).forEach {
+            locationSnapshots.add(LocationSnapshot(racingCars.moveAll()))
         }
-        return locationSnapshots
+        return RacingCarPlayResult(locationSnapshots, racingCars.pickWinners())
```

Comment:
요런 식으로 메서드 체이닝을 해볼 수도 있겠네요 :)
```suggestion
    fun play(): RacingCarPlayResult {
        val racingCars = RacingCars(carNames, moveStrategy)
        
        return (1..numberOfTrials).map { racingCars.moveAll() }
            .map { LocationSnapshot(it) }
            .let { RacingCarPlayResult(it, racingCars.pickWinners()) }
    }
```